src/mpid/ch4/shm/posix/posix_impl.h

```diff
@@ -106,13 +106,15 @@ typedef struct {
 #define MPIDI_POSIX_REQUEST_CREATE_SREQ(sreq_)                  \
     {                                                           \
         (sreq_) = MPIR_Request_create(MPIR_REQUEST_KIND__SEND); \
+        MPIR_ERR_CHKANDSTMT((sreq_)== NULL, mpi_errno, MPIX_ERR_NO_REQ, goto fn_fail, "**nomemreq"); \
         MPIR_Request_add_ref((sreq_));                          \
         (sreq_)->u.persist.real_request   = NULL;               \
     }
 
 #define MPIDI_POSIX_REQUEST_CREATE_RREQ(rreq_)                  \
     {                                                           \
         (rreq_) = MPIR_Request_create(MPIR_REQUEST_KIND__RECV); \
+        MPIR_ERR_CHKANDSTMT((rreq_)== NULL, mpi_errno, MPIX_ERR_NO_REQ, goto fn_fail, "**nomemreq"); \
```

Comment:
I think this should be consistent for all request creations, and not just send and receive requests.

Reply:
Good point. I will update the PR.